src/datasets/synthetic.jl

```diff
@@ -182,8 +182,8 @@ function logit_fun(X, z, setting)
   end
   yprob = log_link.(logit)
   # Compute a random y with probability yprob
-  u = rand(n)
-  return X, ifelse.(yprob .> u, 1, 0)
+  u = rand()
```

Comment:
@pfistfl please check if this change which I made is correct? I guess, you wanted a single float value for u which you wanted to use to compare to the values of yprob